Xamarin.Essentials/Permissions/Permissions.uwp.cs

```diff
@@ -125,18 +125,24 @@ public override Task<PermissionStatus> CheckStatusAsync()
                 return RequestLocationPermissionAsync();
             }
 
+            static PermissionStatus status = PermissionStatus.Unknown;
+
             internal static async Task<PermissionStatus> RequestLocationPermissionAsync()
             {
+                if (status == PermissionStatus.Granted)
```

Comment:
what if the user changes the permission while running?

Reply:
Huh, good point. On the one hand, the docu of [Geolocator.RequestAccessAsync](https://docs.microsoft.com/en-us/uwp/api/windows.devices.geolocation.geolocator.requestaccessasync) says:

> The RequestAccessAsync method prompts the user for permission to access their location. The user is only prompted once (per app). After the first time they grant or deny permission, this method no longer prompts for permission.

This is what led me to believe that buffering might be a good idea (since repeated calls will apparently have no effect?).

OTOH, there is the [Geolocator.StatusChanged](https://docs.microsoft.com/en-us/uwp/api/windows.devices.geolocation.geolocator.statuschanged) event. So, in order to catch changes to the permissions, we should probably react to that event, right?

Reply:
Since we haven't created that event for the other platforms our recommendation is to just query the API before you call a method that requires it. 

I do see your issue though that you state that you want to query on a background thread accordingly, but we check on the UI thread. This probably is a better all up change for .NET MAUI and we should strip away permission checks and make users do it ahead of time.

Reply:
> Since we haven't created that event for the other platforms our recommendation is to just query the API before you call a method that requires it.

... but one could still use the `StatusChanged` event internally in the UWP parts only (to update the buffered permission), right? We don't need it on the other platforms. (I was not suggesting to make the status event part of the public XE Permissions API or anything like that.)

> I do see your issue though that you state that you want to query on a background thread accordingly, but we check on the UI thread. This probably is a better all up change for .NET MAUI and we should strip away permission checks and make users do it ahead of time.

I don't actually think that making sure the permissions are granted is a bad idea. If you strip away the check, then there is no guarantee any more that querying the location will work.

Another alternative might be to simply make sure that the permissions are checked from the main thread (via `MainThread.InvokeOnMainThreadAsync` or similar), instead of throwing an error if this is not the case ...

Reply:
> > Since we haven't created that event for the other platforms our recommendation is to just query the API before you call a method that requires it.
> 
> ... but one could still use the `StatusChanged` event internally in the UWP parts only (to update the buffered permission), right?

I have figured out how to do that. It's rather straightforward. If anyone is interested, I'll be happy to share it.

However, I don't really think it is necessary. The current UWP behavior on the main branch is as follows: If one removes the location permissions in the system settings while the app is running, the next location request simply triggers an exception ("Unable to detect location. Exception: Access is denied." or similar). The permission check does not seem to request the permission from the user again.

Therefore my buffering of the permission status should not be a problem: It is only used to skip the permission request, which is skipped anyway upon a change of the permission status while running. The behavior on my branch is the same as on main in this respect.

Reply:
> > ... but one could still use the `StatusChanged` event internally in the UWP parts only (to update the buffered permission), right?
> 
> I have figured out how to do that. It's rather straightforward. If anyone is interested, I'll be happy to share it.
> 
> However, I don't really think it is necessary.

Sorry, I think I was wrong here. For the case where one starts with a granted permissions, which is buffered upon the first request and then denied in the system settings, everything was working well indeed.

However, I missed to handle the inverted case, where one starts with a buffered `Denied `permission that is changed to `Granted` subsequentially. I fixed that in c097ddd91c1f980ac598cd70bf07db463da56730. And in fact this case also introduces the necessity to handle permission status changes during runtime, as remarked by @jamesmontemagno. This is added in 9e80679fb83e2244bb627d7b8674ea4821dc307b.

@jamesmontemagno I hope you are happy with these additions.